test/libponyc-run/regression-3658/main.pony

```diff
@@ -0,0 +1,14 @@
+primitive Foo[T: (Unsigned & UnsignedInteger[T])]
```

Comment:
Can you add a comment at the top level of this file explaining in general that this is a compile time check so the "additional cases" are covered because this is compile time, not runtime checking?

Reply:
Okay

Reply:
Added in f16a0ea

Reply:
Er, I added the comment locally, not at the top, is that fine?